packages/connect/src/api/firmware/uploadFirmware.ts

```diff
@@ -36,10 +38,31 @@ export const uploadFirmware = async (
     if (device.features.major_version === 1) {
         postConfirmationMessage(device);
         await typedCall('FirmwareErase', 'Success', {});
+
         postProgressMessage(device, 0, postMessage);
-        const { message } = await typedCall('FirmwareUpload', 'Success', {
+
+        let done: boolean = false;
+        let message = typedCall('FirmwareUpload', 'Success', {
             payload,
+        }).then(() => {
+            done = true;
         });
+
+        let i = 0;
+
+        const postProgressMessageIter = async () => {
+            i++;
+            postProgressMessage(device, Math.min(i * 2, 99), postMessage);
```

Comment:
Not sure, maybe not necessary...
I deleted the fake timers entirely because
- newer firmware seems to send the progress info sooner so there is less or no delay
- it reduces code complexity
- it can cause [confusing bugs](https://satoshilabs.slack.com/archives/CKN5C51CJ/p1711363048471019?thread_ts=1709329814.340539&cid=CKN5C51CJ)